parallel.go

```diff
@@ -151,16 +178,28 @@ func (b *blockDesc) decompress() {
 	b.duration = time.Since(start)
 }
 
-func (dc *Decompressor) worker(ctx context.Context, in <-chan *blockDesc, out chan<- *blockDesc) {
+func (dc *Decompressor) worker(ctx context.Context, in <-chan *blockDesc, out chan<- *blockDesc, pool chan struct{}) {
```

Comment:
please set the type of the pool chan to be read only.

Reply:
@cosnicolaou I can't. We use it both reads and writes.